backend/data/processors/programs_processing.py

```diff
@@ -150,15 +167,23 @@ def addDisciplineData(components, item):
                     ):
                         code = major["academic_item_code"]
                         honoursData[code] = major["academic_item_name"]
-                SpecialisationData["Honours"] = honoursData
+                try:
+                    components["SpecialisationData"]["Honours"].update(honoursData)
+                except KeyError:
+                    components["SpecialisationData"]["Honours"] = honoursData
+
             # If item is minor loop through and add data to minors
             if container["vertical_grouping"]["value"] == "undergrad_minor":
                 minorData = {}
                 for minor in container["relationship"]:
                     if minor["academic_item_type"]["value"] == "minor":
                         code = minor["academic_item_code"]
                         minorData[code] = minor["academic_item_name"]
-                SpecialisationData["Minors"] = minorData
+                try:
+                    components["Minors"].update(minorData)
+                except KeyError:
+                    components["Minors"] = minorData
```

Comment:
and here

Reply:
also, are you sure this is supposed to be `components["Minors"]` instead of `components["NonSpecialisationData"]["Minors"]`?